apps/gitness/src/pages/pipeline-list.tsx

```diff
@@ -58,13 +75,35 @@ export default function PipelinesPage() {
             }
           ]
         }))}
+        LinkComponent={LinkComponent}
       />
     )
   }
 
+  //fetch projects api to get the list of projects
+  const { data: projects, isLoading } = useMembershipSpacesQuery({
+    queryParams: { page: 1, limit: 30, sort: 'identifier', order: 'asc' }
+  })
+
+  const projectsItem: Project[] =
+    // @ts-expect-error remove "@ts-expect-error" once type issue for "content" is resolved
+    projects?.content?.map(membership => ({
+      id: `${membership?.space?.id}`,
+      name: membership?.space?.identifier
+    })) || []
+
+  const renderTopBar = () => {
+    //prevent for breaking the pages and wait for the data got response since the data will be undefined from components
+    if (isLoading) {
+      //not sure the design state of the loading, this is temoporary
+      return <div className="w-full h-full text-center">isLoading...</div>
+    }
+    return <TopBarWidget projects={projectsItem || []} />
+  }
+
   return (
     <>
-      <TopBarWidget />
+      {renderTopBar()}
```

Comment:
I'd prefer to use component syntax instead of function syntax

Reply:
move all rendering in components/header-api